### PR TITLE
[WIP] Added webhook support for commentMonitor

### PR DESCRIPTION
**Pull request by geekodour**
This PR removes the ability to read `event.json` and act on its content, instead it runs as a webhook and emits `repository_dispatch` events. 

Signed-off-by: Hrishikesh Barman <hrishikeshbman@gmail.com>

### CODE REVIEW

**Comment by geekodour** [comment]
cc @krasi-georgiev

File: tools/commentMonitor/main.go [hunk 1]
```diff
@@ -14,219 +14,217 @@
...
-					log.Fatalf("%v : couldn't post comment", err)
-				}
-				os.Exit(1)
-			}
-			log.Println("author is a member or collaborator")
+		// Validate regex.
+		err := cmClient.validateRegex()
+		if err != nil {
+			//log.Println(err) // Don't log on failure.
+			http.Error(w, "comment validation failed", http.StatusBadRequest)
```
Comment (krasi-georgiev) [inline tools/commentMonitor/main.go]: with most api's when you send a non 200 ok reply the same request is repeated which is not what we want here. Best to just return 200

I think `validateRegex()` should use bool as the return statment

File: tools/commentMonitor/client.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	commentTemplate string
+}
+
+// Validate comment if regexString provided.
+func (c *commentMonitorClient) validateRegex() error {
+	for _, e := range c.eventMap {
+		c.regex = regexp.MustCompile(e.RegexString)
+		if c.regex.MatchString(c.ghClient.commentBody) {
+			c.commentTemplate = e.CommentTemplate
+			c.eventType = e.EventType
```
Comment (krasi-georgiev) [inline tools/commentMonitor/client.go:43]: why do we need this variable?

**Review by krasi-georgiev — COMMENTED** [review]
also don't forget to update the main prombench readme

File: tools/commentMonitor/client.go (hunk 2 above)
Comment (geekodour) [reply tools/commentMonitor/client.go:43]: These are later used in `createRepositoryDispatch` call in the `extractArgs` function in this file.

Comment (krasi-georgiev) [reply tools/commentMonitor/client.go:43]: yep I sew how it is used now. thanks

**Pull request merged**